src/lib/storage/frame_of_reference_segment.cpp

```diff
@@ -62,14 +62,14 @@ ChunkOffset FrameOfReferenceSegment<T, U>::size() const {
 }
 
 template <typename T, typename U>
-std::shared_ptr<AbstractSegment> FrameOfReferenceSegment<T, U>::copy_using_allocator(
-    const PolymorphicAllocator<size_t>& alloc) const {
-  auto new_block_minima = pmr_vector<T>(_block_minima, alloc);
-  auto new_offset_values = _offset_values->copy_using_allocator(alloc);
+std::shared_ptr<AbstractSegment> FrameOfReferenceSegment<T, U>::copy_using_memory_resource(
+    MemoryResource& memory_resource) const {
+  auto new_block_minima = pmr_vector<T>(_block_minima, &memory_resource);
+  auto new_offset_values = _offset_values->copy_using_memory_resource(memory_resource);
 
   std::optional<pmr_vector<bool>> null_values;
```

Comment:
Not yours, but `auto null_values = std::optional<pmr_vector<bool>>{};`

Not sure if it has a performance impact, but setting `null_values` with a ternary might be better as the optional can be created with the actual vector.